prqlc/prql-compiler/tests/integration/queries.rs

```diff
@@ -8,123 +8,145 @@ use prql_compiler::sql::Dialect;
+    fn run(prql_path: &Path) {
+        let test_name = prql_path.file_stem().unwrap().to_str().unwrap();
+        let prql = fs::read_to_string(prql_path).unwrap();
 
-    let sql = prql_compiler::compile(&prql, &options).unwrap();
+        let target = Target::Sql(Some(Dialect::Generic));
+        let options = Options::default().no_signature().with_target(target);
 
-    with_settings!({ input_file => prql_path }, {
-        assert_snapshot!(snapshot_name, &sql, &prql)
-    });
+        let sql = prql_compiler::compile(&prql, &options).unwrap();
+
+        with_settings!({ input_file => prql_path }, {
+            assert_snapshot!(test_name, &sql, &prql)
+        });
+    }
 }
 
-test_each_path! { in "./prqlc/prql-compiler/tests/integration/queries" as fmt => fmt }
+mod fmt {
+    use super::*;
+
+    test_each_path! { in "./prqlc/prql-compiler/tests/integration/queries" => run }
 
-fn fmt(prql_path: &Path) {
-    let file_stem = prql_path.file_stem().unwrap().to_str().unwrap();
-    let snapshot_name = format!("fmt@{file_stem}");
-    let prql = fs::read_to_string(prql_path).unwrap();
+    fn run(prql_path: &Path) {
+        let test_name = prql_path.file_stem().unwrap().to_str().unwrap();
+        let prql = fs::read_to_string(prql_path).unwrap();
 
-    let pl = prql_compiler::prql_to_pl(&prql).unwrap();
-    let formatted = prql_compiler::pl_to_prql(pl).unwrap();
+        let pl = prql_compiler::prql_to_pl(&prql).unwrap();
+        let formatted = prql_compiler::pl_to_prql(pl).unwrap();
 
-    with_settings!({ input_file => prql_path }, {
-        assert_snapshot!(snapshot_name, &formatted, &prql)
-    });
+        with_settings!({ input_file => prql_path }, {
+            assert_snapshot!(test_name, &formatted, &prql)
+        });
+    }
 }
 
 #[cfg(any(feature = "test-dbs", feature = "test-dbs-external"))]
-#[test]
-fn test_queries_dbs() {
-    use std::{env, fs};
+mod results {
+    use super::*;
+
+    use std::{ops::DerefMut, sync::Mutex};
 
     use anyhow::Context;
-    use insta::{assert_snapshot, glob};
-    use itertools::Itertools;
-    use strum::IntoEnumIterator;
+    use once_cell::sync::Lazy;
+    use prql_compiler::sql::SupportLevel;
+
+    use crate::dbs::{ConnectionCfg, DbConnection};
 
-    use prql_compiler::Options;
-    use prql_compiler::{sql::Dialect, sql::SupportLevel, Target::Sql};
+    static CONNECTIONS: Lazy<Mutex<Vec<DbConnection>>> = Lazy::new(init_connections);
 
-    use crate::dbs::connection::DbConnection;
-    use crate::dbs::IntegrationTest;
+    fn init_connections() -> Mutex<Vec<DbConnection>> {
+        let con_dir = "./tests/integration/dbs/connections";
 
-    let runtime = &*crate::dbs::RUNTIME;
+        let mut connections = Vec::new();
+        for con_file in fs::read_dir(con_dir).unwrap() {
+            let con_file = con_file.unwrap();
+            let con_toml = fs::read_to_string(con_file.path()).unwrap();
+            let con_cfg: ConnectionCfg = toml::from_str(&con_toml)
```

Comment:
I'm fine with this but FWIW I'm generally less keen on breaking things that don't change often / don't require modularity into config files — it makes them further away from the code, means LSPs can't resolve them, adds a bunch more files...

And here, we also have everything defined twice now — structs and a config file — it would be more reasonable if we could make the rust code totally generic and then the DBs were defined in the config file, but we don't get that either.

(I know this isn't immediately impactful, tell me if it feels like peanut-gallery critique, especially given our recent relative contributions...)